src/main/java/com/google/devtools/build/lib/analysis/PlatformConfiguration.java

```diff
@@ -111,4 +114,23 @@ public Label getTargetPlatform() {
     }
     return constraints.build();
   }
+
+  /** Returns true if toolchain resolution debug info should be printed for this toolchain type. */
+  public boolean debugToolchainResolution(Label toolchainType) {
+    return debugToolchainResolution(ImmutableList.of(toolchainType));
+  }
+
+  /**
+   * Returns true if toolchain resolution debug info should be printed for any of these toolchain
+   * types.
+   */
+  public boolean debugToolchainResolution(Collection<Label> toolchainTypes) {
+    if (toolchainTypes.isEmpty()) {
```

Comment:
If this is empty, is there any actual information to debug? i..e what gets printed when this returns true and the input is empty?

Reply:
Even targets with no toolchains will have an execution platform selected (and this is used in some tests).

Ideally we could instead do `--toolchain_resolution_debug=//some/configured:target`, but that would require ConfiguredTargetFunction to make the check and pass a debug flag, thus complicating the ToolchainContextKey even more.